coordinator/src/node/channel.rs

```diff
@@ -0,0 +1,262 @@
+                                settle_transaction, ..
+                            },
+                        ..
+                    }) => (settle_transaction, None),
+                    Channel::SettledClosing(SettledClosingChannel {
+                        settle_transaction,
+                        claim_transaction,
+                        ..
+                    }) => (settle_transaction, Some(claim_transaction)),
+                    channel => {
+                        bail!("DLC channel in unexpected state. dlc_channel = {channel:?}")
+                    }
+                };
+
+                db::dlc_channels::settle_dlc_channel(
+                    &mut conn,
+                    &channel.get_id(),
+                    to_txid_30(settle_transaction.txid()),
+                    claim_transaction.map(|tx| to_txid_30(tx.txid())),
+                )?;
+            }
+            DlcChannelEvent::Closing(_) => {
+                let buffer_transaction = match channel {
+                    Channel::Signed(SignedChannel {
+                        state:
+                            SignedChannelState::Closing {
+                                buffer_transaction, ..
+                            },
+                        ..
+                    }) => buffer_transaction,
+                    Channel::Closing(ClosingChannel {
+                        buffer_transaction, ..
+                    }) => buffer_transaction,
+                    channel => {
+                        bail!("DLC channel in unexpected state. dlc_channel = {channel:?}")
+                    }
+                };
+
+                db::dlc_channels::buffer_dlc_channel(
+                    &mut conn,
+                    &channel.get_id(),
+                    to_txid_30(buffer_transaction.txid()),
+                )?;
+            }
+            DlcChannelEvent::ClosedPunished(_) => {
+                let punish_txid = match channel {
+                    Channel::ClosedPunished(ClosedPunishedChannel { punish_txid, .. }) => {
+                        punish_txid
+                    }
+                    channel => {
+                        bail!("DLC channel in unexpected state. dlc_channel = {channel:?}")
+                    }
+                };
+
+                db::dlc_channels::punish_dlc_channel(
+                    &mut conn,
+                    &channel.get_id(),
+                    to_txid_30(*punish_txid),
+                )?;
+            }
+            DlcChannelEvent::CollaborativeCloseOffered(_) => {
+                let close_transaction = match channel {
+                    Channel::Signed(SignedChannel {
+                        state: SignedChannelState::CollaborativeCloseOffered { close_tx, .. },
+                        ..
+                    }) => close_tx,
+                    channel => {
+                        bail!("DLC channel in unexpected state. dlc_channel = {channel:?}")
+                    }
+                };
```

Comment:
🙃 This is a _good_ opportunity to write a declarative macro using `macro_rules!`, since you've had to use the same pattern of extracting certain fields if the pattern matches.

No need to do it of course, although we might even be able to use one provided by `rust-dlc` if they are public.

Reply:
I looked into it and I didn't see a lot of benefits using the `get_signed_channel_state` macro. The `CollaborativeCloseOffer` handling would be the only one that would benefit from such a macro and still I would have to check first if the channel is a `SignedChannel` so we wouldn't really safe a lot of code.

But maybe a more sophisticated macro could help, but I don't want to spent more time on that right now.

Reply:
Definitely not important to do now (or ever), but I think the pattern would be something like:

```rust
        let mut signed_channel = get_channel_in_state!(self, &channel, Signed, Some(*peer_id))?;
        let close_tx = get_signed_channel_state!(signed_channel, CollaborativeCloseOffered, close_tx)?;
```

This would let us avoid a bunch of verbose explicit `match`es. But we would have to make those macros public anyway.